kusto/ingest/status.go

```diff
@@ -0,0 +1,301 @@
+	"github.com/kylelemons/godebug/pretty"
+	storageuid "github.com/satori/go.uuid"
+)
+
+// StatusCode is the ingestion status
+type StatusCode string
+
+const (
+	// Pending status represents a temporary status.
+	// Might change during the course of ingestion based on the
+	// outcome of the data ingestion operation into Kusto.
+	Pending StatusCode = "Pending"
+	// Succeeded status represents a permanent status.
+	// The data has been successfully ingested to Kusto.
+	Succeeded StatusCode = "Succeeded"
+	// Failed Status represents a permanent status.
+	// The data has not been ingested to Kusto.
+	Failed StatusCode = "Failed"
+	// Queued status represents a permanent status.
+	// The data has been queued for ingestion &  status tracking was not requested.
+	// (This does not indicate that the ingestion was successful.)
+	Queued StatusCode = "Queued"
+	// Skipped status represents a permanent status.
+	// No data was supplied for ingestion. The ingest operation was skipped.
+	Skipped StatusCode = "Skipped"
+	// PartiallySucceeded status represents a permanent status.
+	// Part of the data was successfully ingested to Kusto, while other parts failed.
+	PartiallySucceeded StatusCode = "PartiallySucceeded"
+
+	// StatusRetrievalFailed means the client ran into truble reading the status from the service
+	StatusRetrievalFailed StatusCode = "StatusRetrievalFailed"
+	// StatusRetrievalCanceled means the user canceld the status check
+	StatusRetrievalCanceled StatusCode = "StatusRetrievalCanceled"
+)
+
+// IsFinal returns true if the ingestion status is a final status, or false if the status is temporary
+func (i StatusCode) IsFinal() bool {
+	return i != Pending
+}
+
+// IsSuccess returns true if the status code is a final successfull status code
+func (i StatusCode) IsSuccess() bool {
+	switch i {
+	case Succeeded:
+		return true
+
+	case Queued:
+		return true
+
+	default:
+		return false
+	}
+}
+
+// FailureStatusCode indicates the status of failuted ingestion attempts
+type FailureStatusCode string
+
+const (
+	// Unknown represents an undefined or unset failure state
+	Unknown FailureStatusCode = "Unknown"
+	// Permanent represnets failure state that will benefit from a retry attempt
+	Permanent FailureStatusCode = "Permanent"
+	// Transient represnet a retryable failure state
+	Transient FailureStatusCode = "Transient"
+	// Exhausted represents a retryable failure that has exhusted all retry attempts
+	Exhausted FailureStatusCode = "Exhausted"
+)
+
+// IsRetryable indicates whether there's any merit in retying ingestion
+func (i FailureStatusCode) IsRetryable() bool {
+	switch i {
+	case Transient:
+		return true
+
+	case Exhausted:
+		return true
+
+	default:
+		return false
+	}
+}
+
+// statusRecord is a record containing information regarding the status of an ingation command
+type statusRecord struct {
+	// Status - The ingestion status returned from the service. Status remains 'Pending' during the ingestion process and

```

Comment:
Comment must be complete sentences.  So you cannot do // Status -

Must be of form:

// Status is the ingestion statues returned from the service. ...